src/cockpit/channels/http.py

```diff
@@ -0,0 +1,164 @@
+# This file is part of Cockpit.
+#
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import http.client
+import logging
+import ssl
+import socket
+import threading
+
+from ..channel import Channel
+
+logger = logging.getLogger(__name__)
+
+
+class HttpChannel(Channel):
+    payload = 'http-stream2'
+    connections = {}
```

Comment:
Didn't you say you wanted to drop connection caching?

Reply:
I fixed the "wat" typo. The connection sharing was dropped in a FIXUP, I squashed it now as we are in agreement about not implementing it.